lib/state/ui/reducer.ts

```diff
@@ -21,17 +20,41 @@ const simperiumConnected: A.Reducer<boolean> = (state = false, action) =>
     ? action.simperiumConnected
     : state;
 
-const visiblePanes: A.Reducer<string[]> = (
+const visiblePanes: A.Reducer<Set<string>> = (
   state = defaultVisiblePanes,
   action
 ) => {
-  if ('TAG_DRAWER_TOGGLE' === action.type) {
-    return action.show
-      ? union(state, ['tagDrawer'])
-      : difference(state, ['tagDrawer']);
+  switch (action.type) {
+    case 'App.selectTrash': {
+      return new Set(state).add('trash');
+    }
+    case 'CREATE_NOTE':
+    case 'App.selectTag':
+    case 'App.showAllNotes': {
+      const newSet = new Set(state);
+      newSet.delete('trash');
+      return newSet;
+    }
+    case 'TRASH_TOGGLE':
+      if (state.has('trash')) {
+        const newSet = new Set(state);
+        newSet.delete('trash');
+        return newSet;
+      } else {
+        return new Set(state).add('trash');
+      }
+    case 'TAG_DRAWER_TOGGLE': {
+      if (action.show) {
+        return new Set(state).add('tagDrawer');
+      } else {
+        const newSet = new Set(state);
+        newSet.delete('tagDrawer');
+        return newSet;
+      }
+    }
+    default:
+      return state;
```

Comment:
How do you feel about moving the inclusion/exclusion operations into a callbable function? besides becoming symmetrical we can eliminate the need to manually create new scopes and temporary variables…

```ts
case 'TRASH_TOGGLE':
  return toggle(state, 'trash');

case 'TAG_DRAWER_TOGGLE':
  return action.show
    ? include(state, 'tagDrawer' )
    : exclude(state, 'tagDrawer' );
```